### ISSUE TITLE
Test file found by "testMatch" but somehow nothing found in "Pattern"

### ISSUE THREAD

**Issue by nam-truong-le** [jaredpalmer/tsdx#511]
### Current Behavior

I create a brand new project by running "npx tsdx create" and run "npm test" but no test file is found, see screenshot:

![2020-02-15_13-36-52](https://user-images.githubusercontent.com/1616729/74587903-731d3880-4ff8-11ea-8888-c9905b70c3e9.png)

We can see that one test file was found at the beginning:
```
testMatch: **/__tests__/**/*.[jt]s?(x), **/?(*.)+(spec|test).[tj]s?(x) - 1 match
```

### Expected behavior

Test file should be found.

### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| TSDX           | 0.12.3 |
| TypeScript       | 3.7.5 |
| Browser          | |
| npm/Yarn         | npm 6.13.7 |
| Node             | v12.1.0 |
| Operating System | OSX 10.15.3 |

**Comment by agilgur5** [comment]
Literally got this same issue in one of my projects yesterday, https://github.com/agilgur5/jest-without-globals/pull/2#issuecomment-586417504 . Was extremely confusing and frustrating.

I haven't been able to reproduce it everywhere so I have a feeling this is a bug in the CLI dependency, `sade`. Seems like it just had [some patches released](https://github.com/lukeed/sade/commits/master), so I'm guessing newer TSDX installations grabbed the recent patch while older ones have locked a lower version. Will investigate further

**Comment by agilgur5** [comment]
Oh also another workaround is to pass literally any argument to `tsdx test` and it'll be sliced off instead of `--config`.

This also means to get coverage right now you have to do e.g. `tsdx test --sliceme --coverage`.

A Yarn resolution would be better though.

It might impact other commands too, but not sure as the args are treated differently in `test` due to Jest's [non-existent programmatic API](https://github.com/facebook/jest/issues/5048)

**Comment by agilgur5** [comment]
Confirmed that this does not occur on `sade` v1.7.0. It was introduced in `sade` v1.7.1 (released 3 days ago) due to using a mutative `splice` call on `process.argv` in 2 places, 1 of which is below:
https://github.com/lukeed/sade/blob/4ccffe071af1ceb51b3965539e9ceb8a622ee53e/lib/index.js#L129

It shouldn't mutate `process.argv` -- I'm writing up a PR to `sade` to get that fixed ASAP

**Comment by agilgur5** [comment]
Sent a PR upstream: https://github.com/lukeed/sade/pull/36

Hopefully the root cause is resolved upstream soon.

TSDX could pin `sade`@1.7.0 as well as a stop-gap, but Jared is behind on releases and merges and that would only solve the bug for those who upgraded TSDX to latest, all older versions would get this bug if it's not patched upstream.

**Comment by agilgur5** [comment]
Updating here that https://github.com/lukeed/sade/pull/36 was merged within ~10 hours and released in `sade`@1.7.3 .

Because the bug was upstream, no updates to TSDX itself are necessary. To get this fix in `tsdx` you just need re-install (`npm i tsdx`). The 1.7.2 -> 1.7.3 bump will be in your `package-lock.json` (or `yarn.lock`). All new installs will have the latest version of `sade` by default.

This issue can be closed as it's been resolved.
